packages/peregrine/lib/talons/ProductImageCarousel/useProductImageCarousel.js

```diff
@@ -19,6 +24,22 @@ export const useProductImageCarousel = props => {
         setActiveItemIndex(0);
     }, [images, setActiveItemIndex]);
 
+    useEffect(() => {
+        const urls = images.map(
+            ({ file }) =>
+                `${location.origin}${generateURLFromContainerWidth(
+                    file,
+                    type,
+                    containerWidth
+                )}`
+        );
+        sendMessageToSW(PREFETCH_IMAGES, {
```

Comment:
Yes. If the download is successful we'll have the images ready for future use, if not, it will just continue to be as it is today, which is to fetch when requested.